generate-examples/generate-cert-chain.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+# This script generates a 3-cert ECDSA chain (root -> CA -> issuer) valid for 10 years.
+# Leaf cert uses P-256 (as per the SMART Health Card Framework), CA and root CA use the
+# increasingly stronger P-384 and P-521, respectively (simulatate real PKI).
+
+# directory where intermediate files are kept
+tmpdir=certs
+mkdir -p $tmpdir
+
+# generate self-signed root CA cert
+openssl req -x509 -new -newkey ec:<(openssl ecparam -name secp521r1) -keyout $tmpdir/root_CA.key -out $tmpdir/root_CA.crt -nodes -subj "/CN=SMART Health Card Example Root CA" -days 3650 -config openssl_ca.cnf -extensions v3_ca
```

Comment:
I looked at the output certs in the jwks.json file later and it looks like all 3 certs are signed using ECDSA with SHA-256.

The root cert uses P-521 so the appropriate hashing algorithm for signing using a key of that strength would be SHA-512. Looks like the openssl used to build the samples is defaulting to SHA-256. Depending on openssl version, I believe setting the hashing algorithm for the signature may be as simple as adding -sha512 to the list of openssl arguments.

Since the intermediate is signed using the root CA's P-512 key, it should also be signed using ECDSA with SHA-512.

On the other hand, the end-entity cert should be signed using ECDSA with SHA-384, since the intermediate key used to sign it has a P-384 curve key.

Reply:
I'll use sha-512 for root CA and sha-384 for CA.